### ISSUE TITLE
Error:  undefined method `text' for nil:NilClass

### ISSUE THREAD

**Issue by akshaykmr** [algolia/algoliasearch-jekyll#34]
I have been constantly getting this error, whereas jekyll build works without any errors.

```
Extracting data from reference/03.reference-request/index.html
jekyll 2.5.3 | Error:  undefined method `text' for nil:NilClass
```

**Comment by akshaykmr** [comment]
I'll provide some context.

> index.html

``` html
                <section id="page">
                    <h1>{{ seg4 | split: '_' | join: ' ' | capitalize_all }}</h1>       
                    <div class="content action"> 
                        {% comment %}
                        {% if page.toc %}
                        <div class="panel panel-default">
                            <div class="panel-heading">Contents</div>
                            <div class="panel-body">
                                <div class="toc" id="toc"></div>
                            </div>
                        </div>
                        {% endif %}
                        {% for post in site.posts reversed %}
                        {% if post.url == page.url %} 
                        <!-- Display content and replace [callout] with markup -->
                        {{ post.content | replace:'[callout]','<div class="callout">' | replace:'[callout-danger]','<div class="callout-danger"><i class="title">Important: </i>' | replace:'[callout-info]','<div class="callout-info"><i class="icon-info"></i>' | replace:'[callout-warning]','<div class="callout-warning"><i class="title">Note: </i>' | replace:'[/callout]','</div>' }}
                        {% endif %}
                        {% endfor %}
                   {% endcomment %} 
                    </div>
                </section>
```

Notice the comment tag. I would be getting the above error with this. Now as in the next piece of code If I move the comment tag one level up to include the div(content action). like so 

``` html
                <section id="page">
                    <h1>{{ seg4 | split: '_' | join: ' ' | capitalize_all }}</h1>
                    {% comment %}       
                    <div class="content action">  
                        {% if page.toc %}
                        <div class="panel panel-default">
                            <div class="panel-heading">Contents</div>
                            <div class="panel-body">
                                <div class="toc" id="toc"></div>
                            </div>
                        </div>
                        {% endif %}
                        {% for post in site.posts reversed %}
                        {% if post.url == page.url %} 
                        <!-- Display content and replace [callout] with markup -->
                        {{ post.content | replace:'[callout]','<div class="callout">' | replace:'[callout-danger]','<div class="callout-danger"><i class="title">Important: </i>' | replace:'[callout-info]','<div class="callout-info"><i class="icon-info"></i>' | replace:'[callout-warning]','<div class="callout-warning"><i class="title">Note: </i>' | replace:'[/callout]','</div>' }}
                        {% endif %}
                        {% endfor %}
                    </div>
                    {% endcomment %} 
                </section>
```

Then it works. Also works if I remove the div tag keeping it's inner contents(and removing the comment tag altogether). This behaviour is totally bizarre.

**Comment by pixelastic** [comment]
Thanks for the report, I'll have a look.

Would you have a repository of your whole website so I can replicate the issue on my side? If not I'll try with your code snippets, but this would be easier with the full picture :)

**Comment by akshaykmr** [comment]
Cannot share the repo. I ended up just renaming that div tag to `article`.  😕

**Comment by pixelastic** [comment]
OK, glad you managed to work around the issue.

I have no idea what could cause this, I will try to replicate it.

On Wed, Aug 03, 2016 at 9:48 PM, Akshay Kumar < notifications@github.com [notifications@github.com] > wrote:
Cannot share the repo. I ended up just renaming that div tag to article .

**Comment by pixelastic** [comment]
Hello @akshayKMR, I have tried to replicate your issue again on my side, but can't find a way to trigger it.

I've recently released an updated version of the plugin here: https://github.com/algolia/jekyll-algolia

I will be deprecating this one (not updating it) in the coming weeks. If you ever experience the issue again with the new plugin, feel free to post an issue in the new project and I'll look into it.

Cheers,
